app/src/main/java/com/ojasvi/memify/MainActivity.kt

```diff
@@ -57,27 +61,36 @@ fun MainScreen(modifier: Modifier = Modifier) {
 }
 
 @Composable
-fun ImagePicker(modifier: Modifier = Modifier){
-    var imageUri by remember { mutableStateOf<Uri?>(null)}
-    
-    val imagePickerLauncher = rememberLauncherForActivityResult(contract = ActivityResultContracts.GetContent())
-    { uri: Uri? ->
+fun ImagePicker(modifier: Modifier = Modifier) {
+    var imageUri by remember { mutableStateOf<Uri?>(null) }
+
+    val pickMedia = rememberLauncherForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri ->
+        // Callback is invoked after the user selects a media item or closes the
+        // photo picker.
         imageUri = uri
     }
-    
-    LaunchedEffect(Unit) {
-        imagePickerLauncher.launch("image/*")
-    }
-    
-    
+
+
     Column(
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.Center,
-        modifier = Modifier.fillMaxSize()
-    ){
+        modifier = Modifier.fillMaxSize().padding(20.dp)
+    ) {
+        Button(
+            onClick = { pickMedia.launch(PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)) },
+            modifier = Modifier.fillMaxWidth().height(50.dp)
```

Comment:
Static heights are bad. Can cut the content off on some devices.

Reply:
Updated to use wrapContentHeight() instead